pkg/cmd/server/server.go

```diff
@@ -266,9 +273,19 @@ func newServer(f client.Factory, config serverConfig, logger *logrus.Logger) (*s
 
 	clientConfig, err := f.ClientConfig()
 	if err != nil {
+		cancelFunc()
```

Comment:
I think we should probably `defer cancelFunc()` right after L272 instead of adding it the the various error handling sections

Reply:
Yeah, this was a quick fix for linting complaints, but a `defer` would be better.